atom/browser/browser_mac.mm

```diff
@@ -87,6 +87,23 @@
 void Browser::SetAppUserModelID(const base::string16& name) {
 }
 
+void Browser::SetUserActivity(const std::string& type, const std::map<std::string, std::string>& user_info) {
+  NSString* type_ns = [NSString stringWithUTF8String:type.c_str()];
+  NSUserActivity *user_activity = [[NSUserActivity alloc] initWithActivityType:type_ns];
```

Comment:
The `user_activity`, `user_info_args`, and the array passed to `user_activity.userInfo` should be released after being used, you can put them in a `base::scoped_nsobject` to make them release automatically.